src/pages/ProfileForm/index.tsx

```diff
@@ -98,34 +71,32 @@ function StepForm(): JSX.Element | null {
 
   const isRoutingEnabled = getIsRoutingEnabled();
 
+  /* If the email is in the `denied_domain` list (e.g. personal email addresses) */
   if (isRoutingEnabled && !isEmailDomainAllowed) {
-    ProfileFormState.setState({
-      selectedScenario: Scenario.Error1,
-      step: StepTwo,
-    });
+    return (
+      <Navigate replace to='/summary' state={{ scenario: scenarios.Error1 }} />
```

Comment:
> refactor(Step2): Removed Step2 and useProfileState

I'm curious why you removed your `useProfileState` work and instead are putting it in the react-router state? Is it to make it easier to share across form implementations instead of using zustand?

Reply:
Also, if you want to save any refactors around this for later, no worries. 👍